src/main/java/com/example/bookstoreapp/servive/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.example.bookstoreapp.servive.impl;
+
+import com.example.bookstoreapp.dto.CartItemDto;
+import com.example.bookstoreapp.dto.ShoppingCartDto;
+import com.example.bookstoreapp.mapper.CartItemMapper;
+import com.example.bookstoreapp.mapper.ShoppingCartMapper;
+import com.example.bookstoreapp.model.CartItem;
+import com.example.bookstoreapp.model.ShoppingCart;
+import com.example.bookstoreapp.model.User;
+import com.example.bookstoreapp.repository.CartItemRepository;
+import com.example.bookstoreapp.repository.ShoppingCartRepository;
+import com.example.bookstoreapp.repository.UserRepository;
+import com.example.bookstoreapp.security.CustomUserDetailsService;
+import com.example.bookstoreapp.servive.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Autowired
+    private CartItemMapper cartItemMapper;
+    @Autowired
+    private ShoppingCartMapper shoppingCartMapper;
```

Comment:
why do u use @Autowired if u have @RequiredArgsConstructor

Reply:
> why do u use @Autowired if u have @requiredargsconstructor

It underline with red then. I dont know why.....